remote/RemoteImGuiServer.cpp

```diff
@@ -31,93 +32,88 @@ namespace imgui {
-
-					// Send confirmation
-					SendText("ImInit", 6);
-
-					// Send font texture
-					_sendFontFrame();
-				}
-			}
-			else if (strstr((char *)data, "ImMouseMove")) {
-				int x, y, mouse_left, mouse_right;
-				if (sscanf((char *)data, "ImMouseMove=%d,%d,%d,%d", &x, &y, &mouse_left, &mouse_right) == 4) {
-					_onImMouseMove(x, y, mouse_left, mouse_right);
-				}
-			}
-			else if (strstr((char *)data, "ImMousePress")) {
-				int l, r;
-				if (sscanf((char *)data, "ImMousePress=%d,%d", &l, &r) == 2) {
-					_onImMousePress(l, r);
-				}
-			}
-			else if (strstr((char *)data, "ImMouseWheelDelta")) {
-				float mouseWheelDelta;
-				if (sscanf((char *)data, "ImMouseWheelDelta=%f", &mouseWheelDelta) == 1) {
-					_onImMouseWheelDelta(mouseWheelDelta);
-				}
-			}
-			else if (strstr((char *)data, "ImKeyDown")) {
-				int key, shift, ctrl;
-				if (sscanf((char *)data, "ImKeyDown=%d,%d,%d", &key, &shift, &ctrl) == 3) {
-					_onImKeyDown(key, shift, ctrl);
+	void RemoteImGuiServer::OnMessage(IWebSocketServer::OpCode opcode, const void *data, int size) {
+		switch (opcode) {
+			case IWebSocketServer::Text: {
+				if (size > 0) {
+					const unsigned char * text = static_cast<const unsigned char *>(data);
+					RemoteMessageType messageType = static_cast<RemoteMessageType>(text[0]);
+					_handleMessage(messageType, &text[1], size - 1);
 				}
+				break;
 			}
-			else if (strstr((char *)data, "ImKeyUp")) {
-				int key;
-				if (sscanf((char *)data, "ImKeyUp=%d", &key) == 1) {
-					_onImKeyUp(key);
-				}
+			case IWebSocketServer::Disconnect: {
+				mIsClientActive = false;
+				break;
 			}
-			else if (strstr((char *)data, "ImKeyPress")) {
-				unsigned int key;
-				if (sscanf((char *)data, "ImKeyPress=%d", &key) == 1)
-					_onImKeyPress(key);
+			case IWebSocketServer::Continuation:
+			case IWebSocketServer::Binary:
+			case IWebSocketServer::Ping:
+			case IWebSocketServer::Pong: {
+				// Valid, but unhandled
+				break;
 			}
-			else if (strstr((char *)data, "ImClipboard=")) {
-				_onImClipboard(nullptr, &((char *)data)[strlen("ImClipboard=")]);
+			default: {
+				// Unsupported message type
+				assert(0);
+				break;
 			}
-			break;
-
-		
-			break;
-
-		case IWebSocketServer::Disconnect:
-			mIsClientActive = false;
-			break;
-
-		case IWebSocketServer::Binary:
-		case IWebSocketServer::Ping:
-		case IWebSocketServer::Pong:
-			break;
-		
-		default:
-			assert(0);
-			break;
 		}
 	}
 
+	bool RemoteImGuiServer::_getIsActive() const {
+		return mIsClientActive;
+	}
+
 	void RemoteImGuiServer::_sendFrame(const Frame& frame) {
 		SendBinary(frame.data, frame.size);
 	}
 
-	bool RemoteImGuiServer::_getIsActive() const {
-		return mIsClientActive;
+	void RemoteImGuiServer::_handleMessage(RemoteMessageType messageType, const void * data, int size) {
+		switch (messageType) {
+			case RemoteMessageType::RelayRoomJoined:
+			case RemoteMessageType::RelayRoomUpdate: {
+				// Valid, but unhandled
+				break;
+			}
```

Comment:
These are error cases, right? You should never receive one of these messages unless you are using relay so I would remove them entirely and let them fallthrough to default: